### PR TITLE
Plat 12250 build xcframeworks

### PR DESCRIPTION
**Pull request by kstenerud**
## Goal

This PR adds the ability to build the binary xcframework from the command line.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -89,6 +89,9 @@ build_ios_static: ## Build the static library target
 build_carthage: ## Build the latest pushed commit with Carthage
 	@./scripts/build-carthage.sh
 
+build_xcframework: ## Build as a multiplatform xcframework
```
Comment (twometresteve) [inline Makefile:92]: nit: The Make target calls it xcframework but the script just uses framework.  Maybe less likely to trip people up if we us framework for both?  I can change it on the PR to add it to CI, so long as you agree.

Comment (kstenerud) [reply Makefile:92]: Oh, it's actually building an xcframework, so best to settle on that I guess

Comment (twometresteve) [reply Makefile:92]: Yeah, I don't mind at all which one is used, I'm just likely to forget and get them wrong if they aren't the same - I'll chang eit to xc when I get to it no problem.

Comment (kstenerud) [reply Makefile:92]: Everything should be xcframework now

**Review by twometresteve — APPROVED** [review]

**Pull request merged**